### DISCUSSION TITLE
Regarding srsRAN_4G UE log generation

### DISCUSSION THREAD

**Discussion by SurabhiKarandikar** [srsran/srsRAN_4G#1184] [Q&A]
Greetings.
I am using srsRAN_4G for learning and my current project. While trying to generate log and pcap files for analyzing, I am unable to get information in the ue.log file generated. Moreover, the UE side pcap files are also not getting generated.
I have an observation that the 'srsenb' folder contains [parser.cc, enb_cfg_parser.h, enb_cfg_parser.cc]. However, these files are missing in the srsue folder. Is that the reason behind the issue I am facing?
It would be a great help if you would assist me on this.

Waiting for your reply.

Yours sincerely,
Surabhi Karandikar

**Comment by brendan-mcauliffe** [comment]
Copying my reply on the mailing list here: 

Have you made sure that the logs are enabled in the UE and eNB configuration files? 

As for the 'missing' files, they are specific to the eNB and are not part of the UE source code, so no need to worry about that. 

As an example, if you take a look at the UE example config [here](https://github.com/srsRAN/srsRAN_4G/blob/master/srsue/ue.conf.example): 
- you can enable the pcaps on line 94 by changing `enable = none` to `enable = mac, nas`. 
- you can enable the logs on line 120 by changing `all_level = warning` to `all_level = info`. There are various logging levels available to choose the one that best suits your needs. Debug gives the most info, none gives the least. 
You can enable the logs and pcaps in the eNB in a similar manner to the above. Take a look at the comments in the example configs for more information. 

I should also make sure to run the UE and eNB using `sudo`.

In the future, it is better to use _either_ the mailing list _or_ the discussion forum. Try not to use both as it may lead to a delay in getting your desired answer.
